qiskit_experiments/calibration/management/update_library.py

```diff
@@ -197,19 +197,34 @@ def update(
             CalibrationError: If the experiment is not of the supported type.
         """
         from qiskit_experiments.calibration.rabi import Rabi
+        from qiskit_experiments.calibration.fine_amplitude import FineAmplitude
 
         if angles_schedules is None:
             angles_schedules = [(np.pi, "amp", "xp")]
 
-        if isinstance(exp_data.experiment, Rabi):
-            result = exp_data.analysis_result(result_index)
+        result = exp_data.analysis_result(result_index)
 
+        if isinstance(exp_data.experiment, Rabi):
             freq = result["popt"][result["popt_keys"].index("freq")]
             rate = 2 * np.pi * freq
 
             for angle, param, schedule in angles_schedules:
                 value = np.round(angle / rate, decimals=8)
 
                 cls._add_parameter_value(calibrations, exp_data, value, param, schedule, group)
+
+        elif isinstance(exp_data.experiment, FineAmplitude):
+            d_theta = result["popt"][result["popt_keys"].index("d_theta")]
+
+            for target_angle, param, schedule in angles_schedules:
```

Comment:
You don't assume nonlinearlity of amplitude here?

Reply:
The code here right now only does the right thing if `target_angle` is the same angle as was used in the experiment, so in that sense it is compatible with assuming nonlinearity (if you calibrate xp, this won't update the amplitude the right way for x90p).

Reply:
I'm resolving as I don't see any action item here.